### DISCUSSION TITLE
PrimeFaces chatbot component?

### DISCUSSION THREAD

**Discussion by tshuba** [primefaces/community#430] [PrimeFaces] [answered]
Hello - has there been any consideration in adding a chatbot component to any of the Primefaces libraries?  There are obviously numerous sites at this point that allow one to create a bot and embed it anywhere.... but that also requires a dependency on them.  

Code wise, it's easy to interact with the ChatGPT OpenAI API for example, but it seems like it would be nice have a UI component that would be consistent with other PrimeFaces components instead of creating our own.  This would give the developer full control over the chatbot interaction.  Just a thought.

Cheers.

**Comment by tandraschko** [comment, marked as answer]
Thats just much to much for a UI component framework

**Reply by melloware** [reply]
@tshuba I agree with @tandraschko but I have seen people use the Terminal component https://www.primefaces.org/showcase/ui/misc/terminal/basic.xhtml for similar type of chatbot in the past.  Not the best but closer.

**Reply by tshuba** [reply]
Thank you for those replies...  And I understand your reasoning.  I didn't mean to imply that the UI component would be a fully functional chatbot by any means.  I would still expect all of the necessary programming to be handled by the developer.  What I'm looking for I can accomplish today with a combination of the PF components.  It just requires a little more work. :-)
